series/series.go

```diff
@@ -740,3 +746,186 @@ func (s Series) Quantile(p float64) float64 {
 
 	return stat.Quantile(p, stat.Empirical, ordered, nil)
 }
+
+// Histogram generates a mapping representing a histogram with bins
+// equally spaced binwidth in size.
+// If we did not specify a value for binwidth, i.e. 0 or negative
+// value passed, we compute binwidth ourselves to give a hopefully
+// sane number of bins.
+func (s Series) Histogram(binwidth float64) map[float64]int {
+	if s.elements.Len() == 0 ||
+		s.Type() == String ||
+		s.Type() == Bool {
+		return nil
+	}
+	h := make(map[float64]int)
+	if binwidth == 0 || binwidth < 0 {
+		// For series with lengths less than 30 we use a different
+		// algorithm than for series longer than 30, mainly to avoid
+		// having too few bins in some cases.
+		if s.Len() < 30 {
+			binwidth = (s.Max() - s.Min()) / math.Sqrt(float64(s.Len()))
+		} else {
+			k := math.Cbrt(float64(s.Len()))
+			binwidth = 2 * (s.Quantile(0.75) - s.Quantile(0.25)) / k
+		}
+	}
+	bin := func(n float64) float64 {
+		return math.Floor(n - float64(int(n)%int(binwidth)))
+	}
+	for _, v := range s.Float() {
+		h[bin(v)]++
+	}
+	return h
+}
+
+// Median calculates the middle or median value, as opposed to
+// mean it is not susceptible to being affected by outliers.
+func (s Series) Median() float64 {
+	if s.elements.Len() == 0 ||
+		s.Type() == String ||
+		s.Type() == Bool {
+		return math.NaN()
+	}
+	ix := s.Order(false)
+	newElem := make([]Element, len(ix))
+
+	for newpos, oldpos := range ix {
+		newElem[newpos] = s.elements.Elem(oldpos)
+	}
+
+	// When length is odd, we just take length(list)/2
+	// value as the median.
+	if len(newElem)%2 != 0 {
+		return newElem[len(newElem)/2].Float()
+	}
+	// When length is even, we take middle two elements of
+	// list and the median is an average of the two of them.
+	return (newElem[(len(newElem)/2)-1].Float() +
+		newElem[len(newElem)/2].Float()) * 0.5
+}
+
+func sum(s []float64) float64 {
+	if len(s) > 1 {
+		return s[0] + sum(s[1:])
+	}
+	return s[0]
+}
+
+// Sum calculates sum of given series
+func (s Series) Sum() float64 {
+	if s.elements.Len() == 0 || s.Type() == String {
+		return math.NaN()
+	}
+	return sum(s.Float())
+}
+
+func applyFloat(s1, s2 Series, op int) interface{} {
+	delta := make([]float64, s1.Len())
+	for i := 0; i < s1.Len(); i++ {
+		left := s1.elements.Elem(i).Val().(float64)
+		right := s2.elements.Elem(i).Val().(float64)
+		if math.IsNaN(left) || math.IsNaN(right) {
+			delta[i] = math.NaN()
+			continue
+		}
+		if op == Add {
+			delta[i] = left + right
+		} else if op == Subtract {
+			delta[i] = left - right
+		} // Maybe other operations should be supported
+	}
+	return delta
+}
+
+func applyInt(s1, s2 Series, op int) interface{} {
```

Comment:
You could get rid of the casting by transforming s1 and s2 to float ahead of time using `Series.Int`